pkg/client/dtclient/settings_client.go

```diff
@@ -841,3 +862,106 @@ func (d *SettingsClient) Delete(ctx context.Context, objectID string) error {
 
 	return nil
 }
+
+func (d *SettingsClient) GetPermission(ctx context.Context, objectID string) (PermissionObject, error) {
+	path, err := getPermissionPathWithID(objectID)
+	if err != nil {
+		return PermissionObject{}, err
+	}
+
+	resp, err := coreapi.AsResponseOrError(d.client.GET(
+		ctx,
+		path,
+		corerest.RequestOptions{CustomShouldRetryFunc: corerest.RetryIfTooManyRequests},
+	))
+
+	apiError := coreapi.APIError{}
+	// when the API returns a 404 it means that you don't have permission (no-access)
+	if errors.As(err, &apiError) && apiError.StatusCode == http.StatusNotFound {
+		return PermissionObject{}, nil
+	}
+
+	if err != nil {
+		return PermissionObject{}, fmt.Errorf("failed to get permission: %w", err)
+	}
+
+	var result PermissionObject
+	if err = json.Unmarshal(resp.Data, &result); err != nil {
+		return PermissionObject{}, fmt.Errorf("failed to unmarshal permission response: %w", err)
+	}
+
+	return result, nil
+}
+
+func (d *SettingsClient) UpsertPermission(ctx context.Context, objectID string, permission PermissionObject) error {
+	path, err := getPermissionPathWithID(objectID)
+	if err != nil {
+		return err
+	}
+
+	payload, err := json.Marshal(permission)
+	if err != nil {
+		return fmt.Errorf("failed to marshal permission: %w", err)
+	}
+
+	// In the first sep we try to update the remote object
+	_, err = coreapi.AsResponseOrError(d.client.PUT(
+		ctx,
+		path,
+		bytes.NewReader(payload),
+		corerest.RequestOptions{CustomShouldRetryFunc: corerest.RetryIfTooManyRequests},
+	))
+
+	// Successful create
+	if err == nil {
+		return nil
+	}
+
+	apiError := coreapi.APIError{}
+	// On a 404 we step throw and try to create the object
+	if errors.As(err, &apiError) && apiError.StatusCode != http.StatusNotFound {
+		return fmt.Errorf("failed to update permission: %w", err)
+	}
+
+	// endpoint for POST is different to PUT
+	path = strings.Replace(settingsPermissionAPIPath, "{objectId}", objectID, -1)
+	_, err = coreapi.AsResponseOrError(d.client.POST(
+		ctx,
+		path,
+		bytes.NewReader(payload),
+		corerest.RequestOptions{CustomShouldRetryFunc: corerest.RetryIfTooManyRequests},
+	))
+
+	if err != nil {
+		return fmt.Errorf("failed to create permission: %w", err)
+	}
+
+	return nil
+}
+
+func (d *SettingsClient) DeletePermission(ctx context.Context, objectID string) error {
+	path, err := getPermissionPathWithID(objectID)
+	if err != nil {
+		return err
+	}
+	_, err = coreapi.AsResponseOrError(d.client.DELETE(
+		ctx,
+		path,
+		corerest.RequestOptions{CustomShouldRetryFunc: corerest.RetryIfTooManyRequests},
+	))
+
+	if err != nil {
+		return fmt.Errorf("failed to delete permission object: %w", err)
+	}
+
+	return nil
+}
+
+// When creating a permission we need to point to settingsPermissionAPIPath else we point to settingsPermissionAllUsersAPIPath
+func getPermissionPathWithID(id string) (string, error) {
```

Comment:
You say in the doc that this handles both cases (all users and 'normal' one), but in the end you handle only the all-users case. If you keep it to all-users only, please update the function name